### PR TITLE
Use all gpus available for training

### PR DESCRIPTION
**Pull request by Abecid**
Use all gpus available for training

### CODE REVIEW

File: finetune/adapter.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
 save_interval = 1000
 eval_iters = 100
 log_interval = 1
-devices = 1
+devices = torch.cuda.device_count()
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply finetune/adapter.py:39]: @carmocca see the logic below on line 69

I vote for not doing this in adapter.py for keeping things simple.

File: finetune/full.py [hunk 2]
```diff
@@ -31,7 +31,7 @@
 save_interval = 1000
 eval_iters = 100
 log_interval = 100
-devices = 4
+devices = torch.cuda.device_count()
```
Comment (awaelchli) [inline finetune/full.py:34]:
```suggestion
devices = "auto"
```

File: finetune/lora.py [hunk 3]
```diff
@@ -50,7 +50,8 @@ def main(
     out_dir: str = "out/lora/alpaca",
 ):
 
-    fabric = L.Fabric(accelerator="cuda", devices=1, precision="bf16-true")
+    devices = torch.cuda.device_count()
+    fabric = L.Fabric(accelerator="cuda", devices=devices, precision="bf16-true")
```
Comment (awaelchli) [inline finetune/lora.py:54]:
```suggestion
    fabric = L.Fabric(accelerator="cuda", devices="auto", precision="bf16-true")
```

Comment (awaelchli) [reply finetune/lora.py:54]: @Abecid Did multi-gpu training work with this script and how many did you use? Is the loss convergence comparable to single gpu training?

**Review by awaelchli — COMMENTED** [review]
Thanks for the suggestion @Abecid. In general the changes look good but it won't make much difference because we are keeping the batch size constant globally to keep it easy to reproduce results. In fact, if training fits in a single GPU and torch.cuda.device_count() selects more devices, it may actually be slower because of the reduced per-device batch size and communication overhead. 

I vote for keeping the values as they are and instead advise users to scale both devices and batch size to their setup to maximize efficiency. We could add this info to the how-to guides. Let me know what you think.

File: finetune/adapter.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Borda) [reply finetune/adapter.py:39]:
```suggestion
devices = "auto"
```
